lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -267,19 +267,26 @@ def get_course_assignments(self, model: CourseEnrollment) -> Optional[Dict[str,
             self.context.get('request'),
             include_past_dates=True
         )
-        next_assignment = None
+        next_assignment_all = []
+        next_assignment = []
         past_assignment = []
 
         timezone = get_user_timezone_or_last_seen_timezone_or_utc(model.user)
         for assignment in sorted(assignments, key=lambda x: x.date):
             if assignment.date < datetime.now(timezone):
                 past_assignment.append(assignment)
             else:
-                next_assignment = DateSummarySerializer(assignment).data
-                break
+                if not assignment.complete:
+                    next_assignment_all.append(assignment)
+
+        if next_assignment_all:
+            future_assignment_date = next_assignment_all[0].date.date()
+            next_assignment = [
+                assignment for assignment in next_assignment_all if assignment.date.date() == future_assignment_date
+            ]
```

Comment:
```suggestion
if future_assignments:
        future_assignment_date = future_assignments[0].date.date()
        next_assignments = [
            assignment for assignment in future_assignments if assignment.date.date() == future_assignment_date
        ]
    else:
        next_assignments = []
```
Just to avoid initializing above